ui/component/headerNotificationButton/index.js

```diff
@@ -1,14 +1,26 @@
 import { connect } from 'react-redux';
-import { selectUnseenNotificationCount } from 'redux/selectors/notifications';
-import { doSeeAllNotifications } from 'redux/actions/notifications';
-import { selectUser } from 'redux/selectors/user';
+import { selectNotifications, selectUnseenNotificationCount } from 'redux/selectors/notifications';
+import {
+  doReadNotifications,
+  doSeeNotifications,
+  doDeleteNotification,
+  doSeeAllNotifications,
+} from 'redux/actions/notifications';
+import { selectUser, selectUserVerifiedEmail } from 'redux/selectors/user';
 import NotificationHeaderButton from './view';
 
 const select = (state) => ({
+  notifications: selectNotifications(state),
   unseenCount: selectUnseenNotificationCount(state),
   user: selectUser(state),
+  authenticated: selectUserVerifiedEmail(state),
 });
 
-export default connect(select, {
-  doSeeAllNotifications,
-})(NotificationHeaderButton);
+const perform = (dispatch, ownProps) => ({
+  readNotification: ([id]) => dispatch(doReadNotifications([id])),
+  seeNotification: ([id]) => dispatch(doSeeNotifications([id])),
+  deleteNotification: (id) => dispatch(doDeleteNotification(id)),
```

Comment:
There is no modifications and not using `ownProps`, right?  No need to change (since it requires renaming the variables), but for future can just use the object version for 1:1 mapping.

```
const perform = {
  doReadNotifcations,
  doSeeNotifications,
  ...
};
```

Reply:
I kept the mapping to make it clear that we only pass one id (`notification`) instead of multiple ones (`notifications`), but you're correct, `ownProps` is no longer needed.